sources/packages/backend/apps/api/src/services/application-change-request/application-change-request.service.ts

```diff
@@ -0,0 +1,161 @@
+        changeRequestApplication.student.id,
+        NoteType.Application,
+        note,
+        auditUserId,
+        transactionalEntityManager,
+      );
+      const applicationRepo =
+        transactionalEntityManager.getRepository(Application);
+      if (applicationEditStatus === ApplicationEditStatus.ChangeDeclined) {
+        // Update the application edit status.
+        const updateResult = await applicationRepo.update(
+          {
+            id: applicationId,
+            applicationEditStatus: ApplicationEditStatus.ChangePendingApproval,
+          },
+          {
+            applicationEditStatus: ApplicationEditStatus.ChangeDeclined,
+            modifier: auditUser,
+            updatedAt: currentDate,
+          },
+        );
+        if (!updateResult.affected) {
+          throw new Error(
+            `Application ${applicationId} to assess change not found or not in valid status to be updated.`,
+          );
+        }
+        // End the workflow.
+        await this.workflowClientService.sendApplicationChangeRequestStatusMessage(
+          applicationId,
+          applicationEditStatus,
+        );
+        return;
+      }
+      const previousCompletedApplication =
+        changeRequestApplication.precedingApplication;
+      const previousCompletedApplicationId = previousCompletedApplication.id;
+      const previousCompletedApplicationCurrentAssessment =
+        previousCompletedApplication.currentAssessment;
+      const newApplicationCurrentAssessment =
+        changeRequestApplication.currentAssessment;
+      // Copy the most recent offering id and the student appeal id
+      // from the latest preceding application to the newly approved current application.
+      newApplicationCurrentAssessment.offering = {
+        id: previousCompletedApplicationCurrentAssessment.offering?.id,
+      } as EducationProgramOffering;
+      if (previousCompletedApplicationCurrentAssessment.studentAppeal) {
+        newApplicationCurrentAssessment.studentAppeal = {
+          id: previousCompletedApplicationCurrentAssessment.studentAppeal.id,
+        } as StudentAppeal;
+      }
+      // Update the previously completed application to be in Edited status.
+      await applicationRepo.update(
+        {
+          id: previousCompletedApplicationId,
+        },
+        {
+          applicationStatus: ApplicationStatus.Edited,
+          modifier: auditUser,
+          updatedAt: currentDate,
+        },
+      );
+      const updateResult = await applicationRepo.update(
+        {
+          id: applicationId,
+          applicationEditStatus: ApplicationEditStatus.ChangePendingApproval,
+        },
+        {
+          applicationEditStatus: ApplicationEditStatus.ChangedWithApproval,
+          applicationStatus: ApplicationStatus.Completed,
+          currentAssessment: newApplicationCurrentAssessment,
+          modifier: auditUser,
+          updatedAt: currentDate,
```

Comment:
There are more audit properties which are pending to be updated. 

`applicationStatusUpdatedOn`
`applicationEditStatusUpdatedOn`
`applicationEditStatusUpdatedBy`